File: Data Structures & Algorithms/concatenation-of-array/submission-1.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        ans, i = [], 0
        n = len(nums) * 2

        while n > 0:
            if n == len(nums):
                i = 0
            ans.append(nums[i])
            n -= 1
            i += 1

        return ans

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int Search(int[] nums, int target) {
        // Set up the low and high pointers
        int low = 0;
        int high = nums.Length - 1;

        // While the low and high pointers are not
        while (low <= high) {
            int mid = (low + high) / 2;

            if (nums[mid] == target) {
                return mid;
            } else if (nums[low] <= nums[mid]) {
                if (target > nums[mid] || target < nums[low]) {
                    low = mid + 1;
                } else {
                    high = mid - 1;
                }
            } else {
                if (target < nums[mid] || target > nums[high]) {
                    high = mid - 1;
                } else {
                    low = mid + 1;
                }
            }
        }

        return -1;
    }
}
